### PR TITLE
Auth part is tested

### PR DESCRIPTION
**Pull request by STQT**

### CODE REVIEW

File: smart_city/posts/migrations/0009_code.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 3.2.14 on 2022-08-08 09:21
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
```
Comment (STQT) [inline smart_city/posts/migrations/0009_code.py:9]: Mana shu Code classni yaratib keyin o'chirganizda makemigrations yaratib keyin classni o'chirib yana makemigrations yaratganiz serverga kerak emas))) shunga ikta faylni o'chirib keyin git add qilib yuklab yuboring :). Aytgancha fayllarni o'chirishdan oldin ushbu kodni yozing. 
`python manage.py migrate posts 0008` keyin o'chirib tashlayvering

File: smart_city/posts/models.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 from ckeditor_uploader.fields import RichTextUploadingField
 from django.db import models
+from django.db.models.fields import related
```
Comment (STQT) [inline smart_city/posts/models.py:3]: related ishlatilmayaptimi?

File: smart_city/users/urls.py [hunk 3]
```diff
@@ -4,24 +4,27 @@
...
-    user_update_view,
+    user_update_view, RegisterAPIView, VerifyCodeView,
 )
 from .views import (FacebookLogin, GitHubLogin,GoogleLogin, FacebookConnect, GithubConnect)
 
 app_name = "users"
 urlpatterns = [
+    path('register/',RegisterAPIView.as_view(),name="register"),
+    path('verify/',VerifyCodeView.as_view(),name="verify"),
+
```
[the remark this answers is not in the document]
Comment (Baxtiyor98) [reply smart_city/users/urls.py:15]: Hop

File: smart_city/users/models.py [hunk 4]
```diff
@@ -2,15 +2,34 @@
...
         return reverse("users:detail", kwargs={"username": self.username})
 
+
+
+class Code(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    number = models.CharField(max_length=6,null=True, blank=True)
+
+    def __str__(self):
+        return f"{self.number}"
```
[the remark this answers is not in the document]
Comment (Baxtiyor98) [reply smart_city/users/models.py:26]: Code kiritish registrationda verified qismi

File: smart_city/posts/serializer.py [hunk 5]
```diff
@@ -1,33 +1,87 @@
...
+        depth = 1
+
+class SearchNewsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = News
+        fields = ['id', 'title','theme','tags','description']
+        depth = 1
+
+class SearchArticlesSerializer(serializers.ModelSerializer):
+    class Meta:
```
[the remark this answers is not in the document]
Comment (Baxtiyor98) [reply smart_city/posts/serializer.py:37]: Haa

File: smart_city/posts/models.py (hunk 2 above)
Comment (Baxtiyor98) [reply smart_city/posts/models.py:3]: Men ishlatmadim

**Comment by STQT** [comment]
Tugadi. Tamom

**Pull request merged**